CMakeLists.txt

```diff
@@ -0,0 +1,107 @@
+cmake_minimum_required (VERSION 2.6)
+project (j2v8)
+
+# adding cmake directory for includes
+set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+
+include (BuildUtils)
+include (NodeJsUtils)
+include (Policies)
+
+#-----------------------------------------------------------------------
+# CMAKE OPTIONS
+#-----------------------------------------------------------------------
+option(J2V8_NODE_COMPATIBLE "Build the J2V8 native bridge with Node.js support enabled" ON)
+option(J2V8_BUILD_ONLY_DEBUG_RELEASE "Generate only Debug and Release configurations (exclude RelWithDebInfo and MinSizeRel)" ON)
+
+# set up the module path
+set (CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+# remove the MinSizeRel and RelWithDebInfo configurations
+if (J2V8_BUILD_ONLY_DEBUG_RELEASE)
+#{
+    set (CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "limited configs" FORCE)
+#}
+endif ()
+
+#-----------------------------------------------------------------------
+# DEPENDENCY SETTINGS & DISCOVERY
+#-----------------------------------------------------------------------
+
+# look for dependencies
+find_package(Java)
+
+# overridable settings
+set (J2V8_JDK_DIR ${Java_ROOT} CACHE STRING "Path to the Java JDK dependency")
+set (J2V8_NODEJS_DIR "./node" CACHE STRING "Path to the Node.js dependency")
+
+# get lists of the required Node.js link libraries
+get_njs_libs (${J2V8_NODEJS_DIR} "Debug")
+get_njs_libs (${J2V8_NODEJS_DIR} "Release")
+
+#-----------------------------------------------------------------------
+# INCLUDE DIRECTORIES & SOURCE FILES
+#-----------------------------------------------------------------------
+
+# project include directories
+set (include_dirs
+    ${J2V8_JDK_DIR}/include
+    ${J2V8_JDK_DIR}/include/win32
+    ${J2V8_NODEJS_DIR}
+    ${J2V8_NODEJS_DIR}/src
+    ${J2V8_NODEJS_DIR}/deps/v8
+    ${J2V8_NODEJS_DIR}/deps/v8/include
+)
+
+# project source files
+set (src_files
+    jni/com_eclipsesource_v8_V8Impl.cpp
+    jni/com_eclipsesource_v8_V8Impl.h
+)
+
+source_group("" FILES ${src_files})
+
+#-----------------------------------------------------------------------
+# BUILD SETTINGS & COMPILATION
+#-----------------------------------------------------------------------
+
+link_static_crt()
+
+# create the j2v8 library
+add_library (j2v8 SHARED ${src_files})
+
+# enable Node.js if requested by the set options
+if (J2V8_NODE_COMPATIBLE)
+#{
```

Comment:
I use those just by convention to make it easier for the eye to quickly read where the scope of a if-elif-else block starts and ends. This being said, for the next PR I have removed those where only a single line is inside the branch scope, and they are only used for if-elif-else blocks that contain multiple lines in their respective blocks. Thanks for the feedback